src/java/io/bazel/rulesscala/scalac/BUILD

```diff
@@ -25,7 +25,7 @@ java_binary(
         "//third_party/bazel/src/main/protobuf:worker_protocol_java_proto",
         "@io_bazel_rules_scala//src/java/io/bazel/rulesscala/jar",
         "@io_bazel_rules_scala//src/java/io/bazel/rulesscala/worker",
-        "@io_bazel_rules_scala//src/protobuf/io/bazel/rules_scala:diagnostics_java_proto",
+        "@bazel_tools//src/main/protobuf:diagnostics_java_proto",
```

Comment:
hmm, now that I see I think this might be a problem since it will mean that our users will have to use the latest version of bazel (possibly even the not released one). Maybe we should return the proto file for a few months

Reply:
The problem is that the newest version of bazel will be required nonetheless as it stands. Changes made [in this PR](https://github.com/bazelbuild/bazel/pull/11766) are required to output a Diagnostics File through BEP, since that was not supported in bazel either way.

Reply:
There's a difference between requiring it for people opting in than requiring it for everyone.

Reply:
Then perhaps my opt-in implementation needs to be rechecked, since the changes made to actions on bazel side make it so that an action has an additional parameter. This is why I mistakingly committed the commenting out of the bazel script provided in the tools, it does not compile in earlier bazel versions. Would this be possible to detect and solve?

Reply:
We might be able to condition the starlark code based on bazel version. Do you mean that `diagnostics_file` is a new argument?
We used to check existence of attributes on dictionaries to know if the bazel version is relevant. Maybe there's a better way.

Reply:
Yes it is a new argument, added in the PR I mentioned. I'll try to see if I can I do anything of that sorts then. If you do happen to know a better way, please do share!

Reply:
Is it to check the number of arguments a function gets in starlark? I think that's the only better solution I'm coming up with rather than checking attributes as you mentioned. 
If you can't, do you happen to know which attribute would lead me to understand which bazel version it is being used?

Reply:
Here's an example of doing things conditionally in starlark based on the bazel version: https://github.com/bazelbuild/rules_rust/blob/fdf9655ba95616e0314b4e0ebab40bb0c5fe005c/rust/private/rustc.bzl#L204-L209

Reply:
Thanks for the tip Daniel, was able to use this!